File: BackEnd/src/main/java/com/helpmybrain/service/UsuarioService.java
package com.helpmybrain.service;

import com.helpmybrain.entity.Usuario;

import java.util.List;

public interface UsuarioService {
    Usuario guardarUsuario(Usuario usuario);
    Usuario actualizarUsuario(Usuario usuario);
    void eliminarUsuario(Integer id);
    Usuario obtenerUsuarioPorId(Integer id);
    Usuario obtenerUsuarioPorEmail(String email);
    List<Usuario> obtenerTodosLosUsuarios();
}

File: frontend/src/services/UsuarioService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/helpmybrain/usuarios/';

class UsuarioService {
    obtenerTodosLosUsuarios() {
        return axios.get(API_URL);
    }

    obtenerUsuarioPorId(id) {
        return axios.get(API_URL + id);
    }

    crearUsuario(usuario) {
        return axios.post(API_URL, usuario);
    }

    actualizarUsuario(id, usuario) {
        return axios.put(API_URL + id, usuario);
    }

    eliminarUsuario(id) {
        return axios.delete(API_URL + id);
    }
}

export default new UsuarioService();
